src/cclib/parser/data.py

```diff
@@ -83,67 +84,68 @@ class ccData(object):
-        "geovalues":      numpy.ndarray,
-        "grads":          numpy.ndarray,
-        "hessian":        numpy.ndarray,
-        "homos":          numpy.ndarray,
-        "mocoeffs":       list,
-        "moenergies":     list,
-        "moments":        list,
-        "mosyms":         list,
-        "mpenergies":     numpy.ndarray,
-        "mult":           int,
-        "natom":          int,
-        "nbasis":         int,
-        "nmo":            int,
-        "nocoeffs":       numpy.ndarray,
-        "nooccnos":       numpy.ndarray,
-        "optdone":        bool,
-        "optstatus":      numpy.ndarray,
-        "scancoords":     numpy.ndarray,
-        "scanenergies":   list,
-        "scannames":      list,
-        "scanparm":       list,
-        "scfenergies":    numpy.ndarray,
-        "scftargets":     numpy.ndarray,
-        "scfvalues":      list,
-        "temperature":    float,
-        "vibanharms":     numpy.ndarray,
-        "vibdisps":       numpy.ndarray,
-        "vibfreqs":       numpy.ndarray,
-        "vibirs":         numpy.ndarray,
-        "vibramans":      numpy.ndarray,
-        "vibsyms":        list,
+    Attribute = namedtuple('Attribute', ['type','jsonKey'])
+    _attributes = {
+       "aonames":          Attribute( list,             'names'),
+       "aooverlaps":       Attribute( numpy.ndarray,    'overlaps'),
+       "atombasis":        Attribute( list,             'indices'),
+       "atomcharges":      Attribute( dict,             'atomic charge'),
+       "atomcoords":       Attribute( numpy.ndarray,    'coords'),
+       "atommasses":       Attribute( numpy.ndarray,    'mass'),
+       "atomnos":          Attribute( numpy.ndarray,    'number'),
+       "atomspins":        Attribute( dict,             'spins'),
+       "ccenergies":       Attribute( numpy.ndarray,    'coupled cluster'),
+       "charge":           Attribute( int,              'charge'),
+       "coreelectrons":    Attribute( numpy.ndarray,    'core electrons'),
+       "enthalpy":         Attribute( float,            'enthalpy'),
+       "entropy":          Attribute( float,            'entropy'),
+       "etenergies":       Attribute( numpy.ndarray,    'electronic transitions'),
+       "etoscs":           Attribute( numpy.ndarray,    'oscillator strength'),
+       "etrotats":         Attribute( numpy.ndarray,    'rotatory strength'),
+       "etsecs":           Attribute( list,             'one excited config'),
+       "etsyms":           Attribute( list,             'symmetry'),
+       "freeenergy":       Attribute( float,            'free energy'),
+       "fonames":          Attribute( list,             'orbital names'),
+       "fooverlaps":       Attribute( numpy.ndarray,    'orbital overlap'),
+       "fragnames":        Attribute( list,             'names'),
+       "frags":            Attribute( list,             'atom indices'),
+       "gbasis'":          Attribute( list,             'TBD'),

```

Comment:
Travis CI is failing because of ```gbasis``` when using ```ccData_optdone_bool```. I think that is because there is an extra ```'``` here.